doozerlib/assembly_inspector.py

```diff
@@ -63,12 +76,52 @@ def does_permit(self, issue: AssemblyIssue) -> bool:
                     return True
         return False
 
+    def _check_installed_packages_for_rpm_delivery(self, component: str, component_description: str, rpm_packages: Dict[str, Dict[str, Any]]):
+        """
+        If rpm_deliveries is configured, this checks installed packages in a component
+        follow the stop-ship/ship-ok tagging requirement.
+        See https://issues.redhat.com/browse/ART-6100 for more information.
+
+        :param component: Component name, like `rhcos` or image distgit key
+        :param component_description: Component description, used to include more details for the component in the AssemblyIssue message
+        :param rpm_packages: A dict; key is rpm package name, value is brew build dict
+        """
+        self.runtime.logger.info("Checking installed rpms in %s", component_description)
+        issues: List[AssemblyIssue] = []
+        for package_name, rpm_build in rpm_packages.items():
+            if package_name in self._rpm_deliveries:
+                rpm_delivery_config = self._rpm_deliveries[package_name]
+                self.runtime.logger.info("Getting tags for rpm build %s...", rpm_build['nvr'])
+                tag_names = {tag["name"] for tag in self.brew_session.listTags(build=rpm_build["id"])}
+                # If the rpm is tagged into the stop-ship tag, it is never permissible
+                if rpm_delivery_config.stop_ship_tag and rpm_delivery_config.stop_ship_tag in tag_names:
+                    issues.append(AssemblyIssue(f'{component_description} has {rpm_build["nvr"]}, which has been tagged into the stop-ship tag: {rpm_delivery_config.stop_ship_tag}', component=component, code=AssemblyIssueCode.IMPERMISSIBLE))
+                    continue
+                # For GA releases, the rpm must have explicit "ship-ok" tag
```

Comment:
might want a permissible issue code for pre-GA releases (ECs, RCs), so that we are at least conscious we're going to need a change for GA?

Reply:
We didn't really discuss what to do with pre-GA releases before. Adding a permissible issue sounds good to me.